roles/infra/tasks/main.yml

```diff
@@ -8,6 +8,49 @@
       }
     dest: "{{ terraform_project_path }}/backend.tf"
 
+- block:
+  - name: Set a temporary fact containing terraform_backend_config
+    set_fact:
+      _terraform_backend_config: "{{ terraform_backend_config }}"
+      terraform_backend_config:
+    when:
+      - terraform_backend_config.force_path_style is defined
+      - not disable_s3_force_path_style_checks
+
+  - block:
+    - name: Warn if force_path_style is used in terraform_backend_config
+      fail:
+        msg: >-
+          The terraform_backend_config.force_path_style configuration
+          option is deprecated and terraform_backend_config.use_path_style
+          should be used instead!
+      ignore_errors: true
+
+    - name: Rename force_path_style to use_path_style
```

Comment:
Can't this just be done with something like:

```yaml
set_fact:
  terraform_backend_config: {{ terraform_backend_config | combine({'use_path_style':_path_style}) }}
vars:
  _path_style: terraform_backend_config.force_path_style
when:
  - "'force_path_style' in terraform_backend_config
```

maybe you need to take a dict copy from the old one, if that causes a recursion.

Reply:
Unfortunately then, you end up with both `use_path_style` and `force_path_style` in the backend config, which is not allowed.

This has to be an add/remove (replace) operation rather than just a straight add.